### PR TITLE
Issue1 multiple zeroshot topics per document

### LINKED ISSUE #1: Implement multiple topics per document for zero-shot
https://github.com/semandex/nlp2/issues/201

### PR DESCRIPTION
**Pull request by ianrandman**
Closes #1

Completion is pending on a response to https://github.com/MaartenGr/BERTopic/issues/1967#issuecomment-2165960807.
Not all unit tests pass.
Workflow is any commits from https://github.com/semandex/BERTopic/tree/issue2-simplify-zero-shot will be cherry-picked into this branch, as the branches diverge due to custom changes to BERTopic, but fixes related to zero-shot are necessary.

### CODE REVIEW

**Review by chirag4semandex — APPROVED** [review]
Dont understand the logic much since not really aware of the requirement but reviewed changes for syntactical changes and it looks good.

File: bertopic/_bertopic.py [hunk 1]
```diff
@@ -4132,21 +4131,43 @@ def _reduce_to_n_topics(self, documents: pd.DataFrame) -> pd.DataFrame:
 
         # Cluster the topic embeddings using AgglomerativeClustering
         if version.parse(sklearn_version) >= version.parse("1.4.0"):
-            cluster = AgglomerativeClustering(self.nr_topics - self._outliers, metric="precomputed", linkage="average")
+            cluster = AgglomerativeClustering(self.nr_topics - self._outliers, metric="precomputed", linkage="complete")
         else:
-            cluster = AgglomerativeClustering(self.nr_topics - self._outliers, affinity="precomputed", linkage="average")
+            cluster = AgglomerativeClustering(self.nr_topics - self._outliers, affinity="precomputed", linkage="complete")
```
Comment (djmakwana) [inline bertopic/_bertopic.py]: both clauses are same?

**Review by djmakwana — APPROVED** [review]
How will we maintain versioning with our patches?

File: bertopic/_bertopic.py (hunk 1 above)
Comment (ianrandman) [reply bertopic/_bertopic.py]: The parameter is `metric` vs `affinity`.

**Comment by ianrandman** [comment]
> How will we maintain versioning with our patches?

We will use a 4th digit after the current BERTopic version. An initial version of this will be `0.16.2.0`. This will be updated when #1 is merged into the BERTopic repo.

**Comment by djmakwana** [comment]
@ianrandman I do not want to add more changes to this PR. Let us create "our_main" branch to hold all our _changes_ on top of BERTopic `master` branch.

**Pull request merged**